core/feconf.py

```diff
@@ -24,11 +24,28 @@
 
 from core.constants import constants
 
-from typing import Dict, List, Union
+from typing import Dict, List
 from typing_extensions import TypedDict
 
-CommandType = (
-    Dict[str, Union[str, List[str], Dict[str, Union[str, List[str]]]]])
+
+# This TypedDict is used only to create the TypedDict below, because we want
+# to mark some attributes as optional.
```

Comment:
Let's not do this. Instead, add allowed_values = {} and deprecated_values = {} to all of the dicts so that they have a consistent format and you don't need to mark anything as optional. In general if we want a strict type system, it is probably better to reduce this sort of branching.

@vojtechjelinek thoughts?

Reply:
Right, that makes more sense, thanks. I focused on some smaller parts of this PR and didn't realize that this is only used for constants that are simple to modify.

Reply:
@seanlip @vojtechjelinek I am not getting where to add `allowed_values = {}` and `deprecated_values = {}`. You mentioned all of the dicts, what do you mean by that?

Reply:
@div-yam In all the dicts where we use this type variable and either `allowed_values` or `deprecated_values` is not filled you can just set it to `{}` and then remove the `total=False` from here. Basically, you need to modify the code so that the MyPy checks pass even when `total=True` here.

Reply:
Can you now remove the `total=False`?